cmake/deprecated.cmake

```diff
@@ -0,0 +1,42 @@
+#######
+# Check for deprecated arguments, provide warning, and set the
+# corresponding option.
+#
+# This file is being added in support of https://github.com/Unidata/netcdf-c/pull/2895 and may eventually
+# no longer be necessary and removed in the future. 
+# 
+#######
+
+set(DEPR_OPT "Warning! Deprecated Options used. Please migrate your build system as follows:" CACHE INTERNAL "" FORCE)
+
+function(check_depr_opt arg)
+
+    if(DEFINED ${arg})
+        #message(STATUS "arg: ${arg} - ${${arg}}")
+        set(val ${${arg}})
+        #MESSAGE("val: ${val}")
+        message(WARNING "${arg} is deprecated and will be removed. Please use NETCDF_${arg} in the future")
+        set(NETCDF_${arg} ${val} CACHE INTERNAL "" FORCE)
```

Comment:
Avoid using `set(FORCE)`. This would overwrite the help message and the cache type, making it hard for user to navigate the cache file. Use either plain `set` (in this case include `PARENT_SCOPE` or `return`), or get the original help message from `get_property(__help_string CACHE ${variable} PROPERTY HELPSTRING)` and append a useful message to it